wpengine/lostromos
# Adding Custom Resource Watcher

# What Are We Doing Here

This commit adds our custom resource watcher. This commit also includes an example controller that just prints out that the CR was added/updated/deleted. This controller is useful for testing and for showing an example of how to implement your own custom controller.

## How to Verify

1. Check out this PR
2. Setup kubectl against a cluster (minikube works just fine)
2. `kubectl apply -f test-data/crd.yml`
3. `kubectl apply -f test-data/cr_things.yml`
4. `go run main.go start --crd-group stable.nicolerenee.io --crd-name characters`
    - See that it prints out that `thing1` and `thing2` were added
5. In another shell `kubectl apply -f test-data/cr_nemo.yml`
    - See that it prints out that `nemo` was added
6. `kubectl edit character nemo` and change a field
    - See that it prints out that `nemo` was changed
7. `kubectl delete -f test-data/cr_nemo.yml`
    - See that it prints out that `nemo` was deleted
8. `kubectl delete -f test-data/cr_things.yml`
     - See that it prints out that `nemo` was deleted
9. Thats it. You can stop the process and `kubectl delete -f test-data/crd.yml` to cleanup the rest of the test data.
